(base) ayumi@ayumi-XPS-8930:~/Desktop/tmp/riscv-gnu-toolchain$ git diff
diff --git a/binutils b/binutils
--- a/binutils
+++ b/binutils
@@ -1 +1 @@
-Subproject commit b51c2fec1da205ea3e7354cbb3e253018d64873c
+Subproject commit b51c2fec1da205ea3e7354cbb3e253018d64873c-dirty
(base) ayumi@ayumi-XPS-8930:~/Desktop/tmp/riscv-gnu-toolchain$ git diff --cached --submodule
(base) ayumi@ayumi-XPS-8930:~/Desktop/tmp/riscv-gnu-toolchain$ cd binutils/
(base) ayumi@ayumi-XPS-8930:~/Desktop/tmp/riscv-gnu-toolchain/binutils$ git diff
diff --git a/include/opcode/riscv-opc.h b/include/opcode/riscv-opc.h
index 88b8d7ff595..84f32c7616a 100644
--- a/include/opcode/riscv-opc.h
+++ b/include/opcode/riscv-opc.h
@@ -2518,6 +2518,11 @@
 #define CSR_VL 0xc20
 #define CSR_VTYPE 0xc21
 #define CSR_VLENB 0xc22
+
+/* Custom Instruction */
+#define MATCH_CUSTOM 0x700b
+#define MASK_CUSTOM 0xfe00707f
+
 #endif /* RISCV_ENCODING_H */
 #ifdef DECLARE_INSN
 DECLARE_INSN(slli_rv32, MATCH_SLLI_RV32, MASK_SLLI_RV32)
diff --git a/opcodes/riscv-opc.c b/opcodes/riscv-opc.c
index 2f9945aa930..f46c0076bc6 100644
--- a/opcodes/riscv-opc.c
+++ b/opcodes/riscv-opc.c
@@ -237,7 +237,7 @@ match_c_slli64 (const struct riscv_opcode *op, insn_t insn)
 }
 
 /* This is used for both srli and srai.  This requires a non-zero shift.
-   A zero rd is not possible.  */
+R  A zero rd is not possible.  */
 
 static int
 match_srxi_as_c_srxi (const struct riscv_opcode *op, insn_t insn)
@@ -1825,6 +1825,9 @@ const struct riscv_opcode riscv_opcodes[] =
 {"hsv.w",       0, INSN_CLASS_H, "t,0(s)", MATCH_HSV_W, MASK_HSV_W, match_opcode, INSN_DREF|INSN_4_BYTE },
 {"hsv.d",      64, INSN_CLASS_H, "t,0(s)", MATCH_HSV_D, MASK_HSV_D, match_opcode, INSN_DREF|INSN_8_BYTE },
 
+/* Custom Instruction */
+{"custom",      0, INSN_CLASS_I, "d,s,t", MATCH_CUSTOM, MASK_CUSTOM, match_opcode, 0 },
+
 /* Terminate the list.  */
 {0, 0, INSN_CLASS_NONE, 0, 0, 0, 0, 0}
 };

